assets/src/edit-story/components/draganddrop/index.js

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+/**
+ * Internal dependencies
+ */
+import { useUploader } from '../../app/uploader';
+import { ReactComponent as UploadIcon } from './icons/upload.svg';
+
+const DragandDropComponent = styled.div`
+  min-width: 100%;
+  min-height: 100%;
+`;
+const DragandDropOverContent = styled.div``;
+
+const DragandDropOverLayWrapper = styled.div`
+  width: 100%;
+  height: 100%;
+  background-color: rgba(0, 0, 0, 0.6);
+  position: absolute;
+  top: 56px;
+  left: 0;
+  z-index: 999;
+`;
+const Heading = styled.h4`
+  color: ${({ theme }) => theme.colors.fg.v1};
+  margin: 0;
+`;
+const Text = styled.p`
+  color: ${({ theme }) => theme.colors.fg.v1};
+`;
+
+const DragandDropOverLay = styled.div`
+  position: absolute;
+  top: 45%;
+  text-align: center;
+  width: 100%;
+`;
+
+function DragandDrop({ children }) {
+  const [isDragging, setIsDragging] = useState(false);
+  const { uploadFile } = useUploader();
+
+  const disableDefaults = (evt) => {
+    evt.preventDefault();
+    evt.stopPropagation();
+  };
+
+  const onDragEnter = (evt) => {
+    disableDefaults(evt);
+    setIsDragging(true);
+  };
+
+  const onDropHandler = (evt) => {
+    disableDefaults(evt);
+    const dt = evt.dataTransfer;
+    let files = dt.files;
+    files = [...files];
+    files.forEach(uploadFile);
+    setIsDragging(false);
+  };
+
+  return (
+    <DragandDropComponent
+      onDragStart={disableDefaults}
+      onDragOver={disableDefaults}
+      onDragLeave={disableDefaults}
+      onDragEnter={onDragEnter}
```

Comment:
For `onDragEnter`, `onDragOver` and `onDrop` - we need to check the media type, etc, to decide whether we want to allow the drop, right?

Reply:
This check happens [here](https://github.com/google/web-stories-wp/blob/f8689299965987595e229d9cb2bbd764858da01e/assets/src/edit-story/app/uploader/useUploader.js#L66). We are awaiting #232 to do this properly.

Reply:
Hmm. Typically you'd want to ignore an unsupported data altogether. E.g. it's usually:

```
if (isValidType()) {
  event.preventDefault();
}
```

This is done so that the other handlers could potentially handle this higher up the DOM. But I wonder if your way is actually is better for our use case since the default browser action is to open whatever is dropped on it. It's so bad that maybe matching the drop target and throwing an error is actually better. Ok. Let's do it this the way you have for now.

Reply:
I think you drag a file or files in here, we should show an error message and not let the browser perform it's default behavior.

Reply:
👍